Sources/Queues/JobData.swift

```diff
@@ -29,6 +29,9 @@ public struct JobData: Codable {
         queuedAt: Date,
         attempts: Int = 0
     ) {
+        precondition(maxRetryCount >= 0)
+        precondition(attempts >= 0)
```

Comment:
I see you're rejecting non-positive values here.
Is it worth it considering it could be surprising to some people?
I'd say let it be what it used to be, just to make sure it doesn't break someone's illogical code.

Reply:
Negative values for either of these parameters (both of which were historically managed by `queues` internally in any event) have always been invalid and caused undefined behavior; these preconditions just make the failure have _defined_ behavior.

Reply:
Still worth it to consider `assert`s 🤔.
You two decide.

Reply:
I'd say an assert is probably safer for anyone doing something stupid like managing the queues separately